simpleclient/src/main/java/io/prometheus/client/LabelValueSanitizer.java

```diff
@@ -0,0 +1,5 @@
+package io.prometheus.client;
+
+public interface LabelValueSanitizer {
+    String[] sanitize(String... labelValue);
```

Comment:
I think it would be better to sanitize only a single label value, not the whole array. For example, you could register a sanitizer like this:
```java
metric = Gauge.build()
        .name("nonulllabels")
        .help("help")
        .labelNames("path", "status)
        .labelValueSanitizer("path", myPathSanitizer)
        .register(registry);
```
and then the `myPathSanitizer` would only be called for the path label value, but not for an array of all label values.

Reply:
Depends on what use case you have in mind.

What I have in mind is that you have a specific label that you want to transform (for example stripping a unique user ID from an HTTP path). For that, it would be good to implement a specific `pathTransformer` and apply it to the `path` label. Advantage is: You don't need to magically know at which position in the array the `path` label is. And if `path` is passed as a label to multiple different metrics, you can reuse the transformer, even if the `path` is on different positions for each metric.

The array as parameter only makes sense if you want to do exactly the same transformation for all labels. This would work for replacing `null` with `"null"`, but for every other use case I think it's better if you know which label you want to transform.